### PR TITLE
refactor(lib): merge `rt`, `geometry` and `scene` submodules into `geometry`

### LINKED ISSUE #497: [FEATURE] Merge `rt`, `geometry` and `scene` submodules into one single submodule
### Terms

- [x] Checked the [existing issues](https://github.com/jeertmans/DiffeRT/issues?q=is%3Aissue+label%3Aenhancement) to see if my suggestion has not already been suggested;

### Description

Since we will likely not support anything else than triangle-based scenes in the near future, and that those three submodules are all referring to purely geometric operations (i.e., not related the EM fields), it would be simpler to merge all those modules into one, e.g., named `geometry` or `geom` for short?

Here is a suggested new layout:

```
differt/src/differt/
├── conftest.py
├── em
│   ├── __init__.py
│   ├── _antenna.py
│   ├── _constants.py
│   ├── _fresnel.py
│   ├── _interaction_type.py
│   ├── _material.py
│   ├── _utd.py
│   └── _utils.py
├── geometry
│   ├── __init__.py
│   ├── _mesh.py                  # Triangles, shapes, etc.
│   ├── _paths.py                 # Ray paths
│   ├── _scene.py                 # Core scene definition
│   ├── _sionna.py                # External integration
│   ├── _solvers.py               # High-level path solvers
│   ├── _solver_fermat.py         # (Previously rt/_fermat.py)
│   ├── _solver_image_method.py   # (Previously rt/_image_method.py)
│   ├── _utils.py                 # Merged geometry, rt, and scene utils
│   ├── scenes/
│   └── scenes.lock               
├── __init__.py
├── plotting
│   ├── __init__.py
│   ├── _core.py
│   └── _utils.py
├── plugins
│   ├── __init__.py
│   ├── deepmimo.py
│   └── _deepmimo_types.py
├── py.typed
├── utils.py
└── _version.py
```

A similar, updated layout should be proposed for `differt-core`.

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jeertmans**
Closes #497.

### CODE REVIEW

File: differt/src/differt/geometry/_utils.py [hunk 1]
```diff
@@ -680,13 +685,15 @@ def viewing_frustum(
             >>> with reuse("plotly") as fig:
             ...     tx = jnp.array([0.0, 0.0, 0.0])
             ...     key = jax.random.key(1234)
-            ...     draw_markers(tx.reshape(-1, 3), labels=["tx"], showlegend=False)
+            ...     _ = draw_markers(tx.reshape(-1, 3), labels=["tx"], showlegend=False)
```
Comment (jeertmans) [inline differt/src/differt/geometry/_utils.py]: I don't think that we should promote this pattern, so let's revert back to not using any variable assignment. If needed, let's skip doctests. Applies here an everywhere else.

File: differt/src/differt/geometry/_utils.py [hunk 2]
```diff
@@ -680,13 +685,15 @@ def viewing_frustum(
...
             ...     for mesh in Mesh.box(with_top=True).translate(tx).iter_objects():
             ...         key, key_color = jax.random.split(key, 2)
             ...         color = r, g, b = jax.random.randint(key_color, (3,), 0, 256)
             ...         center = mesh.bounding_box.mean(axis=0)
-            ...         mesh = mesh.translate(5 * (center - tx)).set_face_colors(color)
-            ...         mesh.plot(opacity=0.5)
+            ...         mesh = mesh.translate(5 * (center - tx)).set_face_colors(
```
Comment (jeertmans) [inline differt/src/differt/geometry/_utils.py:693]: Maybe we should document the difference between [1,0] rgba and [0,255] RGBA, as we maybe support both.

File: differt/tests/test_integration.py [hunk 3]
```diff
@@ -29,7 +27,7 @@ def test_ray_casting() -> None:
     knot_mesh = o3d.data.KnotMesh()
     o3d_mesh = o3d.io.read_triangle_mesh(knot_mesh.path).translate([50, 20, 10])
 
-    o3d_mesh = o3d.t.geometry.Mesh.from_legacy(o3d_mesh)
+    o3d_mesh = o3d.t.geometry.TriangleMesh.from_legacy(o3d_mesh)
```
Comment (jeertmans) [inline differt/tests/test_integration.py]: I don't remember why we even needed to do this, but we should maybe remove this line entirely.

Comment (jeertmans) [reply differt/tests/test_integration.py]: https://www.open3d.org/docs/release/python_api/open3d.t.geometry.TriangleMesh.html#open3d.t.geometry.TriangleMesh.from_legacy

File: CHANGELOG.md [hunk 4]
```diff
@@ -58,6 +58,8 @@ with one *slight* but **important** difference:
 - **Breaking change**: Mask and intersection type arrays of path classes are no longer optional (by <gh-user:jeertmans>, in <gh-pr:491>).
 - **Breaking change**: Renamed `TriangleMesh` to {class}`Mesh<differt.geometry.Mesh>` and `TriangleScene` to {class}`Scene<differt.scene.Scene>` across the codebase. Deprecated aliases are provided for backwards compatibility, raising a `DeprecationWarning` when used (by <gh-user:jeertmans>, in <gh-issue:496>).
```
Comment (jeertmans) [inline CHANGELOG.md]:
```suggestion
```
